sdks/python/src/opik/runtime_config.py

```diff
@@ -0,0 +1,90 @@
+"""
+Runtime configuration for Opik tracing.
+
+This module enables dynamic enabling or disabling of Opik tracing at runtime
+without requiring application restarts or code edits. It acts as a thin,
+thread-safe wrapper around a global flag, falling back to the static
+configuration (`OpikConfig.track_disable`) when no runtime override is set.
+"""
+
+from __future__ import annotations
+
+import threading
+from typing import Optional
+
+
+class _RuntimeConfig:
+    """Thread-safe runtime configuration manager for Opik tracing state."""
+
+    def __init__(self) -> None:
+        self._lock = threading.RLock()
+        self._tracing_active: Optional[bool] = None  # None ⇒ fallback to config
+
+    # ---------------------------------------------------------------------
+    # Public helpers
+    # ---------------------------------------------------------------------
+    def set_tracing_active(self, active: bool) -> None:
+        """Override global tracing state.
+
+        Args:
+            active: ``True`` to enable tracing, ``False`` to disable.
+        """
+        with self._lock:
+            self._tracing_active = active
+
+    def reset_to_config_default(self) -> None:
+        """Clear runtime override so static config is respected again."""
+        with self._lock:
+            self._tracing_active = None
+
+    def is_tracing_active(self) -> bool:
+        """Return current tracing state.
+
+        If a runtime override was set via :func:`set_tracing_active`, that value
+        is returned. Otherwise, we fall back to ``not OpikConfig.track_disable``.
+        If the configuration cannot be loaded for any reason, tracing remains
+        enabled by default (fail open).
+        """
+        with self._lock:
+            if self._tracing_active is not None:
+                return self._tracing_active
+
+        # Fallback lookup happens outside the lock to avoid import deadlocks
+        try:
```

Comment:
Let's cache this check, otherwise it will be too expensive to make on every decorated function run. (OpikConfig instantiation includes reading the config file).